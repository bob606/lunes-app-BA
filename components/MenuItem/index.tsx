import {
  React,
  styles,
  View,
  Text,
  Pressable,
  Image,
  Arrow,
  COLORS,
  IMenuItemProps,
} from './imports';

const MenuItem = ({
  selected,
  onPress,
  icon,
  title,
  children,
}: IMenuItemProps) => {
  const itemStyle = selected ? styles.clickedContainer : styles.container;
  const itemTitleStyle = selected ? styles.clickedItemTitle : styles.itemTitle;

  return (
    <Pressable style={itemStyle} onPress={onPress}>
      <View style={styles.left}>
        <Image source={{uri: icon}} style={styles.icon} />

        <View>
          <Text style={itemTitleStyle} testID="title">
            {title}
          </Text>
          {children}
        </View>
      </View>

      <Arrow
        fill={selected ? COLORS.lunesRedLight : COLORS.lunesBlack}
        testID="arrow"
      />
    </Pressable>
  );
};

export default MenuItem;
