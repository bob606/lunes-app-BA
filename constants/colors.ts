export const COLORS = {
  lunesBlack: '#000e38',
  lunesWhite: '#f9fafb',
  lunesGreyMedium: '#5b616a',
  lunesGreyDark: '#3a4049',
  white: '#ffffff',
  lunesBlackUltralight: '#e0e4ed',
  lunesFunctionalIncorrectDark: '#ff5252',
};
