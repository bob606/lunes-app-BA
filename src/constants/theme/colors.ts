export const COLORS = {
  primary: '#000e38',
  backgroundAccent: '#ffffff',
  background: '#f9fafb',
  text: '#3a4049',
  textSecondary: '#5b616a',
  placeholder: '#777e91',
  disabled: '#e0e4ed',
  exerciseProgressIndicator: '#20c5b4',
  containedButtonSelected: '#3d4662',
  buttonSelectedSecondary: '#f57f7a',
  overlay: 'rgba(0, 14, 56, 0.8)',
  functionalCorrect: '#0adb75',
  functionalAlmostCorrect: '#ffbb4a',
  functionalIncorrect: '#ff5252',
  audioIconHighlight: '#F1635F',
  audioIconSelected: '#ffa3a3',
  shadow: 'rgba(0, 0, 0, 0.6)',
  articleNeutral: '#72f399',
  articlePlural: '#eee12d',
  articleFeminine: '#faa7a7',
  articleMasculine: '#8cc8f3'
}

export type Color = typeof COLORS[keyof typeof COLORS]
