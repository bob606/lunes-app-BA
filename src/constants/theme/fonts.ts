
export const FONTS = {
  contentFontBold: 'SourceSansPro-SemiBold',
  contentFontRegular: 'SourceSansPro-Regular'
}