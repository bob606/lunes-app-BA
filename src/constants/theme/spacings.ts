import { widthPercentageToDP as wp } from 'react-native-responsive-screen'

export const SPACINGS_PLAIN = {
  xxs: wp('0.5%'), // 4
  xs: wp('1%'), // 8
  sm: wp('2%'), // 16
  md: wp('3%'), // 24
  lg: wp('4%'), // 32Â®
  xl: wp('5%'), // 40
  xxl: wp('6%'), // 48
}

// These pixels are calculated for a 800px height mobile screen
export const SPACINGS = {
  xxs: `${SPACINGS_PLAIN.xxs}px`, // 16px
  xs: `${SPACINGS_PLAIN.xs}px`, // 8px
  sm: `${SPACINGS_PLAIN.sm}px`, // 16px
  md: `${SPACINGS_PLAIN.md}px`, // 24px
  lg: `${SPACINGS_PLAIN.lg}px`, // 32px
  xl: `${SPACINGS_PLAIN.xl}px`, // 40px
  xxl: `${SPACINGS_PLAIN.xxl}px`, // 48px
}
