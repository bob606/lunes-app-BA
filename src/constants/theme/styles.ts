export const STYLES = {
  opacity: {
    min: 0.2,
    max: 1,
  },
}
