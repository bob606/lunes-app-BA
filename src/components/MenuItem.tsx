import React, { ReactElement } from 'react'
import { View, Pressable } from 'react-native'
import { Arrow } from '../../assets/images'
import { COLORS } from '../constants/theme/colors'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
import styled from 'styled-components/native'

const ItemStyle = styled(Pressable)`
  justify-content: space-between;
  padding: 17px 8px 17px 16px;
  margin: 0px 16px 8px 16px;
  flex-direction: row;
  align-items: center;
  margin-left: ${(prop: IMenuItemProps) => (prop.selected ? wp('5%') : 16)}px;
  margin-right: ${(prop: IMenuItemProps) => (prop.selected ? wp('5%') : 16)}px;
  background-color: ${(prop: IMenuItemProps) => (prop.selected ? COLORS.lunesBlack : COLORS.white)};
  border-color: ${COLORS.white};
  border-width: ${(prop: IMenuItemProps) => (!prop.selected ? 1 : 0)}px;
  border-style: solid;
  border-radius: ${(prop: IMenuItemProps) => (!prop.selected ? 2 : 0)}px;
`
const ItemTitleStyle = styled.Text`
  font-size: ${wp('5%')}px;
  font-weight: 600;
  letter-spacing: 0.11px;
  margin-bottom: 2px;
  font-family: 'SourceSansPro-SemiBold';
  color: ${(prop: IMenuItemProps) => (prop.selected ? COLORS.white : COLORS.lunesGreyDark)};
`
const Icon = styled.Image`
  justify-content: center;
  margin-right: 10px;
  width: ${wp('7%')}px;
  height: ${wp('7%')}px;
`
const Left = styled.View`
  flex-direction: row;
  align-items: center;
`

export interface IMenuItemProps {
  selected: boolean
  onPress?: () => void
  icon?: string
  title?: string
  children: ReactElement[] | string | undefined
}

const MenuItem = ({ selected, onPress, icon, title, children }: IMenuItemProps): JSX.Element => {
  return (
    <ItemStyle onPress={onPress} selected={selected}>
      <Left>
        <Icon source={{ uri: icon }} />
        <View>
          <ItemTitleStyle selected={selected} testID='title'>
            {title}
          </ItemTitleStyle>
          {children}
        </View>
      </Left>
      <Arrow fill={selected ? COLORS.lunesRedLight : COLORS.lunesBlack} testID='arrow' />
    </ItemStyle>
  )
}
export default MenuItem
