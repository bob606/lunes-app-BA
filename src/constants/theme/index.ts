import { COLORS } from './colors'
import { FONTS } from './fonts'
import { SPACINGS } from './spacings'
import { STYLES } from './styles'

export interface Theme {
  colors: typeof COLORS
  fonts: typeof FONTS
  spacings: typeof SPACINGS
  styles: typeof STYLES
}

const theme: Theme = {
  colors: COLORS,
  fonts: FONTS,
  spacings: SPACINGS,
  styles: STYLES,
}

export default theme
