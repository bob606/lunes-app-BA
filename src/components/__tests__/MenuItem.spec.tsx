import { fireEvent, render } from '@testing-library/react-native'
import { shallow } from 'enzyme'
import React from 'react'
import { Text } from 'react-native'

import { COLORS } from '../../constants/theme/colors'
import wrapWithTheme from '../../testing/wrapWithTheme'
import MenuItem, { IMenuItemProps } from '../MenuItem'

describe('Components', () => {
  describe('MenuItem', () => {
    const defaultMenuItemProps: IMenuItemProps = {
      selected: false,
      icon: '',
      title: 'MenuItemTitle',
      children: <Text>description</Text>,
      onPress: () => {}
    }

    it('should call onPress event', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        onPress: jest.fn()
      }

      const { getByText } = render(<MenuItem {...menuItemProps} />, { wrapper: wrapWithTheme })
      expect(menuItemProps.onPress).not.toHaveBeenCalled()
      const element = getByText('MenuItemTitle')
      fireEvent.press(element)
      expect(menuItemProps.onPress).toHaveBeenCalled()
    })

    it('should display title passed to it', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        title: 'Menu item title'
      }

      const component = shallow(<MenuItem {...menuItemProps} />, { wrappingComponent: wrapWithTheme })
      expect(component.find('[testID="title"]').props().children).toBe(menuItemProps.title)
    })

    it('should render children passed to it', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.contains(menuItemProps.children)).toBe(true)
    })

    it('should render black arrow icon when selected is false', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.find('[testID="arrow"]').props().fill).toBe(COLORS.lunesBlack)
    })

    it('should render red arrow icon when selected is true', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        selected: true
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.find('[testID="arrow"]').props().fill).toBe(COLORS.lunesRedLight)
    })
  })
})
