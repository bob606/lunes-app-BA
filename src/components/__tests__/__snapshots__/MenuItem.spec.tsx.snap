// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components MenuItem renders correctly across screens 1`] = `
<Styled(Pressable)
  onPress={[Function]}
  selected={false}
>
  <Styled(View)>
    <Styled(Image)
      source={
        Object {
          "uri": "",
        }
      }
    />
    <View>
      <Styled(Text)
        selected={false}
        testID="title"
      />
      <Text>
        description
      </Text>
    </View>
  </Styled(View)>
  <SvgrURL
    fill="#000e38"
    testID="arrow"
  />
</Styled(Pressable)>
`;
